sklearn/linear_model/ridge.py

```diff
@@ -206,7 +228,8 @@ def _solve_svd(X, y, alpha):
 
 def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
                      max_iter=None, tol=1e-3, verbose=0, random_state=None,
-                     return_n_iter=False, return_intercept=False):
+                     return_n_iter=False, return_intercept=False,
+                     X_scale=None, X_offset=None):
```

Comment:
You will need to document X_scale and X_offset. This is the reasons that the tests are breaking.

Reply:
These are "internal" variables that need to be propagated down to the solver. They should not be exposed to public API. As discussed with @agramfort I will introduce a new private `_ridge_regression` function that will have X_scale and X_offset, and  `ridge_regression` that will hide them.